### PR TITLE
ActionMenu v2: Add a note about note using it for selection

### PR DESCRIPTION
**Pull request by siddharthkp**
- [x] Added note
- [x] Change the title to ActionMenu v2 to match ActionList v2 in component status page

### Screenshots

Update: Changed into a Note based on @yaili's recommendation. Shows up at the end of examples.

Note that says "Use ActionMenu to choose an action from a list. If you’re looking for single or multiple selection, use DropdownMenu or SelectPanel instead."

### Merge checklist

- [x] Added/updated documentation

### CODE REVIEW

**Comment by colebemis** [comment]
Hmm, this feels like info that should go in the design guidelines (primer.style/design). But I understand that folks might not look at the design guidelines before choosing a component. @yaili any ideas here?

**Comment by siddharthkp** [comment]
> But I understand that folks might not look at the design guidelines before choosing a component

Yes! A tiny amount of duplication to discourage people from using ActionMenu for everything. Also adding warning/error in the code to push towards the right component.

File: docs/content/drafts/ActionMenu2.mdx [hunk 1]
```diff
@@ -240,6 +240,20 @@ render(<Example />)
 | :--------- | :------------------- | :-----: | :-------------------------------- |
 | children\* | `React.ReactElement` |    -    | Required. Accepts a single child. |
 
+<Note variant="warning">
+
+<DoDontContainer>
+  <Do>Use ActionMenu to choose an action from a list.</Do>
+  <Dont>
+    <div>
+      Don't use ActionMenu with for single or multiple selection. Use <a href="/DropdownMenu">DropdownMenu</a> or{' '}
```
Comment (pksjce) [inline docs/content/drafts/ActionMenu2.mdx:249]: remove the `with` ?

**Review by pksjce — APPROVED** [review]

File: docs/content/drafts/ActionMenu2.mdx (hunk 1 above)
Comment (siddharthkp) [reply docs/content/drafts/ActionMenu2.mdx]: that's embarrassing! fixed.

**Comment by siddharthkp** [comment]
Updated to be a note instead of a Do/Don't, much cleaner now! thanks @yaili 

[image](https://user-images.githubusercontent.com/1863771/145562417-e592398d-183d-4ad0-acb9-96b7041be7e1.png)

**Review by yaili — APPROVED** [review]
👍🏼

**Pull request merged**